test/sql/aggregate/qualify/test_qualify.test

```diff
@@ -0,0 +1,188 @@
+----
+Worcester	2019-01-12	98963
+Boston		2019-01-12	486134
+
+query TTR
+SELECT * FROM (SELECT plant, date, avg(mwh) OVER (PARTITION BY plant ORDER BY date ASC RANGE BETWEEN INTERVAL 3 DAYS PRECEDING AND INTERVAL 3 DAYS FOLLOWING) AS avgmwh FROM power 
+ORDER BY plant, avgmwh DESC) QUALIFY row_number() OVER (PARTITION BY plant ORDER BY avgmwh DESC) = 1
+----
+Boston		2019-01-11	532466.666667
+Worcester	2019-01-11	104823.833333
+
+# QUALIFY with condition on ALIAS
+query II
+SELECT b, SUM(a) AS sum FROM test GROUP BY b QUALIFY row_number() OVER (PARTITION BY b) >= 1 AND sum < 20 ORDER BY b;
+----
+21	12
+
+# expression in qualify
+statement ok
+SELECT b, SUM(a) AS sum FROM test GROUP BY b QUALIFY row_number() OVER (PARTITION BY b) > sum * 10;
+----
+
+query ITI
+SELECT * FROM qt QUALIFY row_number() OVER (PARTITION BY b ORDER BY c) = (SELECT max(c) FROM qt)
+----
+2	A	2
+4	B	2
+
+statement ok
+CREATE TABLE tenk1 (unique1 int4, unique2 int4, two int4, four int4, ten int4, twenty int4, hundred int4, thousand int4, twothousand int4, fivethous int4, tenthous int4, odd int4, even int4, stringu1 varchar, stringu2 varchar, string4 varchar)
+
+statement ok
+insert into tenk1 values (8800,0,0,0,0,0,0,800,800,3800,8800,0,1,'MAAAAA','AAAAAA','AAAAxx'), (1891,1,1,3,1,11,91,891,1891,1891,1891,182,183,'TUAAAA','BAAAAA','HHHHxx'), (3420,2,0,0,0,0,20,420,1420,3420,3420,40,41,'OBAAAA','CAAAAA','OOOOxx'), (9850,3,0,2,0,10,50,850,1850,4850,9850,100,101,'WOAAAA','DAAAAA','VVVVxx'), (7164,4,0,0,4,4,64,164,1164,2164,7164,128,129,'OPAAAA','EAAAAA','AAAAxx'), (8009,5,1,1,9,9,9,9,9,3009,8009,18,19,'BWAAAA','FAAAAA','HHHHxx'), (5057,6,1,1,7,17,57,57,1057,57,5057,114,115,'NMAAAA','GAAAAA','OOOOxx'), (6701,7,1,1,1,1,1,701,701,1701,6701,2,3,'TXAAAA','HAAAAA','VVVVxx'), (4321,8,1,1,1,1,21,321,321,4321,4321,42,43,'FKAAAA','IAAAAA','AAAAxx'), (3043,9,1,3,3,3,43,43,1043,3043,3043,86,87,'BNAAAA','JAAAAA','HHHHxx')
+
+# cume_dist
+query I
+SELECT unique1 FROM tenk1 QUALIFY cast(cume_dist() OVER (PARTITION BY four ORDER BY ten)*10 as integer) = 5 order by four, ten
+----
+6701
+4321
+1891
+
+# first_value
+query I
+SELECT unique1 FROM tenk1 QUALIFY first_value(ten) OVER (PARTITION BY four ORDER BY ten) = 1 order by four, ten
+----
+6701	
+4321	
+5057	
+8009	
+1891	
+3043	
+
+# lead
+query I
+SELECT unique1 FROM tenk1 qualify lead(ten * 2, 1, -1) OVER (PARTITION BY four ORDER BY ten) = -1 order by four, ten
+----
+7164	
+8009	
+9850	
+3043	
+
+# in the order of WINDOW clause
+statement ok
+SELECT * FROM exam WINDOW w AS (ORDER BY mark) QUALIFY row_number() OVER w = 1
+
+# must appear WINDOW clause before QUALIFY clause
+statement error
+SELECT * FROM exam QUALIFY row_number() OVER w = 1 WINDOW w AS (ORDER BY mark)
+
+# QUALIFY clause requires at least one window function to be specified in at least one of the SELECT column list or the filter predicate of the QUALIFY clause
+statement error
+SELECT b, avg(a) AS avga FROM test GROUP BY b QUALIFY avga > 10
```

Comment:
Could you add some tests:
* Using QUALIFY in (correlated) subqueries?
* Using QUALIFY in a view that is persistently stored (see e.g. `test/sql/storage/wal/wal_view_storage.test` for an example of such a test)?
* Using QUALIFY in the subquery of a macro function that is persistently stored?

Reply:
Okay, I will add more test cases as you said.